reference-implementation/test/transform-stream.js

```diff
@@ -207,67 +187,59 @@ test('TransformStream: by default, closing the writable waits for transforms to
   setTimeout(() => {
     t.equal(rsClosed, false, 'readable is not closed after a tick');
 
-    ts.writable.closed.then(() => {
-      t.equal(ts.writable.state, 'closed', 'writable becomes closed eventually');
+    writer.closed.then(() => {
+      // TODO: Is this expectation correct?
       t.equal(rsClosed, true, 'readable is closed at that point');
     })
     .catch(e => t.error(e));
   }, 0);
 });
 
 test('TransformStream: by default, closing the writable closes the readable after sync enqueues and async done', t => {
-  t.plan(3);
-
   const ts = new TransformStream({
-    transform(chunk, enqueue, done) {
+    transform(chunk, done, enqueue) {
       enqueue('x');
       enqueue('y');
       setTimeout(done, 50);
     }
   });
 
-  ts.writable.write('a');
-  ts.writable.close();
-  t.equal(ts.writable.state, 'closing', 'writable is closing');
-
-  ts.writable.closed.then(() => {
-    t.equal(ts.writable.state, 'closed', 'writable becomes closed eventually');
+  const writer = ts.writable.getWriter();
+  writer.write('a');
+  writer.close();
 
+  writer.closed.then(() => {
     return readableStreamToArray(ts.readable).then(chunks => {
       t.deepEqual(chunks, ['x', 'y'], 'both enqueued chunks can be read from the readable');
+      t.end();
     });
   })
   .catch(e => t.error(e));
 });
 
 test('TransformStream: by default, closing the writable closes the readable after async enqueues and async done', t => {
-  t.plan(3);
-
   const ts = new TransformStream({
-    transform(chunk, enqueue, done) {
+    transform(chunk, done, enqueue) {
```

Comment:
Oh, very interesting.

I guess I am still hopeful we can get rid of done by using a promise return. But I don't remember where our discussion landed there. We'll figure it out.